fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt

```diff
@@ -0,0 +1,235 @@
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private const val ANIMATION_DURATION_MILLIES = 2000
+
+/**
+ * Download indicator.
+ *
+ * @param text The download indicator text to be displayed.
+ * @param modifier [Modifier] to be applied to the layout.
+ * @param contentDescription to be applied to the layout.
+ * @param backgroundColor The background [Color] of the download indicator.
+ * @param textColor [Color] to apply to the download indicator text.
+ * @param icon Optional [Painter] used to display an [Icon] before the download indicator text.
+ * @param iconContentDescription to be applied to the icon.
+ * @param iconModifier to be applied to the icon.
+ * @param iconTint to be applied to the icon.
+ */
+@Composable
+fun DownloadIndicator(
+    text: String,
+    modifier: Modifier = Modifier,
+    contentDescription: String? = null,
+    backgroundColor: Color = FirefoxTheme.colors.actionPrimary,
+    textColor: Color = FirefoxTheme.colors.textActionPrimary,
+    icon: Painter? = null,
+    iconContentDescription: String? = null,
+    iconModifier: Modifier = Modifier,
+    iconTint: Color = FirefoxTheme.colors.iconActionPrimary,
+) {
+    BaseDownloadIndicator(
+        text = text,
+        textColor = textColor,
+        modifier = modifier,
+        buttonContentDescription = contentDescription,
+        icon = icon,
+        backgroundColor = backgroundColor,
+        iconModifier = iconModifier.then(
+            Modifier.wrapContentSize(),
+        ),
+        iconContentDescription = iconContentDescription,
+        iconTint = iconTint,
+    )
+}
+
+/**
+ * Icon for Download indicator.
+ *
+ * @param modifier [Modifier] to be applied to the icon layout.
+ * @param icon [Painter] used to be displayed.
+ * @param tint to be applied to the icon.
+ * @param iconContentDescription Optional contentDescription for the icon.
+ * */
+@Composable
+fun DownloadIconIndicator(
+    modifier: Modifier = Modifier,
+    icon: Painter,
+    tint: Color = FirefoxTheme.colors.iconActionPrimary,
+    iconContentDescription: String? = null,
+) {
+    BaseLoadingIndicator(
+        icon = icon,
+        modifier = modifier.then(
+            Modifier
+                .rotate(rotationAnimation())
+                .semantics {
+                    if (iconContentDescription != null) {
+                        contentDescription = iconContentDescription
+                    }
+                },
+        ),
+        tint = tint,
+    )
+}
+
+@Composable
+private fun BaseDownloadIndicator(
+    text: String,
+    textColor: Color,
+    modifier: Modifier = Modifier,
+    buttonContentDescription: String? = null,
+    icon: Painter? = null,
+    backgroundColor: Color,
+    iconModifier: Modifier = Modifier,
+    iconTint: Color,
+    iconContentDescription: String? = null,
+) {
+    // Required to detect if font increased due to accessibility.
+    val fontScale: Float = LocalConfiguration.current.fontScale
+
+    Button(
+        enabled = false,
```

Comment:
Just leaving a note for reference - I think `enabled = false,` is the correct state  based on this [conversation](https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD?node-id=7221:186934&mode=design#571221606) -> "not sure it should stay a button, but if so, it should be programmatically `disabled`". 

The TalkBack response for the button seemed as expected for `DownloadIndicatorPreview` (besides the progress bar issue mentioned elsewhere).